catalog/plugin/catalog-plugin-query-monitor-plugin/src/main/java/ddf/catalog/querymonitor/QueryMonitorPluginImpl.java

```diff
@@ -0,0 +1,170 @@
+        return true;
+    }
+
+    public QueryMonitorPluginImpl() {
+        LOGGER.info("QueryMonitor started");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean removeActiveSearch(UUID uniqueID) {
+        if (uniqueID == null) {
+            LOGGER.warn("Can't remove active search with null ID.");
+            return false;
+        }
+        if (activeSearchTable == null) {
+            LOGGER.warn("Can't remove active search. Search not found in table.");
+            return false;
+        }
+        if (activeSearchTable.remove(uniqueID) != null) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Method that is implemented for {@link PreFederatedQueryPlugin}. Uses the given {@link Source}
+     * and {@link QueryRequest} information to create a new {@link ActiveSearch} to add to the ActiveSearchTable.
+     *
+     * @param source {@link Source} that corresponds to source the search is querying
+     * @param input  {@link QueryRequest} that corresponds to request generated when a user queried the source
+     * @return {@link QueryRequest} that was given as a parameter with updated property information corresponding
+     * to the {@link ActiveSearch}'s {@link UUID}
+     */
+
+    @Override
+    public QueryRequest process(Source source, QueryRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        try {
+            if (source == null) {
+                LOGGER.warn("Source given is null.");
+            }
+            UUID uniqueID = UUID.randomUUID();
+            String cqlQuery = null;
+            if (input == null) {
+                LOGGER.warn("QueryRequest given to queryMonitor was null!");
+            } else {
+
+                cqlQuery = input.toString();
+                if (cqlQuery.equals("unknownCQLQuery")) {
+                    LOGGER.warn("CQL Query not set as expected in QueryRequest");
+                } else {
+                    int filterStartIndex = cqlQuery.indexOf("filter=[");
+                    int endOfQueryIndex = cqlQuery.indexOf(" ],startIndex");
+                    if (filterStartIndex == -1 || endOfQueryIndex == -1) {
+                        LOGGER.warn(
+                                "Unable to determine start and end of query terms. Using entire queryRequest instead.");
+                    } else {
+                        cqlQuery = cqlQuery.substring(filterStartIndex + 9, endOfQueryIndex);
+                    }
+                }
+                input.getProperties()
+                        .put("GUID", uniqueID);
+            }
+            addActiveSearch(new ActiveSearch(cqlQuery, source, uniqueID));
+        } catch (Exception e) {
+            LOGGER.warn(
+                    "Exception in queryPlugin monitor. Table may not contain entry for search that caused the exception. Query will continue past queryMonitorPlugin as usual.");
```

Comment:
like jason said above, probably want to refrain from catching `Exception` and favor specific exceptions. 
Also for this Logger you could pass the exception to it as well `LOGGER.warn("........", e)`. Maybe a more concise message would be something like `Unable to add ActiveSearch, Query may not be added to Table`

and upon typing that I realized this is also a fairly large try/catch block. So much that you can't speicifically say what went wrong in the catch block when you log (specifically whether table **may** contain entry. This could also mean the api you are calling should be more clear about the exceptions it throws when something has added or something hasn't been added) Try to keep your try/catch blocks short and concise as possible, trying to wrap as little logic as possible so that you can react to more specific exceptions aptly and also log more supportive error messages .